pkg/client/leaderelection/resourcelock/endpointslock.go

```diff
@@ -60,15 +68,21 @@ func (el *EndpointsLock) Create(ler LeaderElectionRecord) error {
 	if err != nil {
 		return err
 	}
-	el.e, err = el.Client.Core().Endpoints(el.EndpointsMeta.Namespace).Create(&v1.Endpoints{
```

Comment:
Is Endpoints missed in federation clientset? You can generate it by adding `endpoints` at this [line](https://github.com/kubernetes/kubernetes/blob/master/hack/update-codegen.sh#L66). I'm not sure if the generated client will actually work, you might need to register it to the federation's `scheme` as well, but it's worth a try.